### PR TITLE
Add procCallback to spawn and exec functions

### PR DESCRIPTION
**Pull request by PSanetra**
This PR adds a callback with the process object to the spawn function.

Use case: I want to access the stdin of the child process somehow. This is possible via the callback.

### CODE REVIEW

**Comment by PSanetra** [comment]
I have added another commit which extends the `exec` function, too, and also introduces a special `execWithStdin` operator, which executes a command with a string event as stdin input.

File: src/operators.ts [hunk 1]
```diff
@@ -182,3 +185,18 @@ export function throwIfStderr<T>(pattern: string | RegExp) {
...
+    return source.pipe(
+      mergeMap(input =>
+        exec(command, options, proc => {
+          proc.stdin?.write(input);
+          proc.stdin?.end();
+        })
+      )
+    );
+  };
+}
```
Comment (johnny-mh) [inline src/operators.ts:202]: I didn't even think about this. That's a good idea. 👍

**Review by johnny-mh — COMMENTED** [review]
Thank you for your interest in this repository. I made a few comments. If you give me feedback, I'll check it and release a new version. 🙇‍♂️

**Comment by PSanetra** [comment]
@johnny-mh thank you for the quick review! I have resolved those mentioned issues.

**Comment by johnny-mh** [comment]
@PSanetra I just released `v3.1.1`. thank you for your contributing. please let me know anytime if you have any issue.

**Pull request merged**